### ISSUE TITLE
name removed from `body` in `.api_get_message_body`

### ISSUE THREAD

**Issue by LiNk-NY** [Bioconductor/AnVIL#37]
Hi Martin, @mtmorgan 

I am getting an error in `cBioPortalData` due to this behavior. 

When the length of the `body0` / `body` argument is one, the name gets stripped via `body[[1]]`. 

Function: 
https://github.com/Bioconductor/AnVIL/blob/5689b875d06e8b580a87ec40720aa6a6d453031e/R/api.R#L107-L135

Inputs:

```r
op_def <- list(tags = "Gene Panels", summary = "Fetch gene panel data", 
    operationId = "fetchGenePanelDataInMultipleMolecularProfilesUsingPOST", 
    consumes = "application/json", produces = "application/json", 
    parameters = list(list(name = "molecularProfileIds", `in` = "body", 
        type = "array", description = NULL, items = list(type = "string")), 
        list(name = "sampleMolecularIdentifiers", `in` = "body", 
            type = "array", description = NULL, items = list(
                `$ref` = "#/definitions/SampleMolecularIdentifier"))), 
    responses = list(`200` = list(description = "OK", schema = list(
        type = "array", items = list(`$ref` = "#/definitions/GenePanelData")))),
    deprecated = FALSE, path = "/gene-panel-data/fetch", action = "post")

body0 <- list(molecularProfileIds = c("acc_tcga_rppa", "acc_tcga_gistic"))
```

Call: 

```r
body <-  .api_get_message_body(op_def, body0)
```

Expected return: 

```r
jsonlite::toJSON(body0)
# {"molecularProfileIds":["acc_tcga_rppa","acc_tcga_gistic"]} 
```

Actual return:

```r
.api_get_message_body(op_def, body0)
# ["acc_tcga_rppa", "acc_tcga_gistic"] 
```

Possible patch? 

```diff
diff --git a/R/api.R b/R/api.R
index 5560e25..1fef0fb 100644
--- a/R/api.R
+++ b/R/api.R
@@ -110,23 +110,22 @@
     if (identical(op_def$consumes, "multipart/form-data")) {
         json <- body
     } else {
-        if (length(body) == 1L)
-            body <- body[[1]]
-        ## unbox?
-        name <- vapply(op_def$parameters, `[[`, character(1), "name")
-        type <- vapply(op_def$parameters, function(elt) {
-            type <- elt$type
-            if (is.null(type)) {
-                ## FIXME: recurse into $ref
-                NA_character_
-            } else {
-                type
+        if (length(body) > 1L) {
+            name <- vapply(op_def$parameters, `[[`, character(1), "name")
+            type <- vapply(op_def$parameters, function(elt) {
+                type <- elt$type
+                if (is.null(type)) {
+                    ## FIXME: recurse into $ref
+                    NA_character_
+                } else {
+                    type
+                }
+            }, character(1))
+            for (nm in names(body)) {
+                idx <- match(nm, name)
+                if (type[idx] %in% c("string", "number", "integer", "boolean"))
+                    body[[nm]] <- unbox(body[[nm]])
             }
-        }, character(1))
-        for (nm in names(body)) {
-            idx <- match(nm, name)
-            if (type[idx] %in% c("string", "number", "integer", "boolean"))
-                body[[nm]] <- unbox(body[[nm]])
         }
         json <- jsonlite::toJSON(body, pretty = TRUE)
     }
```

Reports: 

https://bioconductor.org/checkResults/devel/bioc-LATEST/cBioPortalData/
https://bioconductor.org/checkResults/release/bioc-LATEST/cBioPortalData/

**Comment by LiNk-NY** [comment]
Did you get a chance to look at this Martin @mtmorgan ? Thanks!

**Comment by mtmorgan** [comment]
Yes. From `git blame` I see

```
commit 98c4b2d2799ed355672dabe657346f7252ef9dbd
Author: Martin Morgan <Martin.Morgan@RoswellPark.org>
Date:   Fri Nov 27 15:00:56 2020 -0500

    don't name length-1 body arguments

    - fixes #30
```

so this is 'by design'. A workaround is to enclosing the argument in `list()`.

**Comment by LiNk-NY** [comment]
Thanks Martin!
Perhaps there should be a distinction between what goes in the `path` and what goes in the `body`? 
But this would be a separate issue.
